core/server/master/src/main/java/alluxio/master/block/RegisterStreamObserver.java

```diff
@@ -36,7 +38,11 @@
 public class RegisterStreamObserver implements StreamObserver<RegisterWorkerPRequest> {
   private static final Logger LOG = LoggerFactory.getLogger(RegisterStreamObserver.class);
 
-  private WorkerRegisterContext mContext;
+  @GuardedBy("this")
+  @SuppressFBWarnings(value = "IS_FIELD_NOT_GUARDED")
+  // Context is initialized on the 1st request so later requests are guaranteed to see the context
+  // Locking is applied on init and cleanup
+  private volatile WorkerRegisterContext mContext;
```

Comment:
i figure the `volatile` is missing as this field is seen by multiple threads @elega please take another look thanks!